asciidoctor-epub-example not working with Java17

Works fine with Java 11, but with Java17 and no kindlegen fails.

```
        <maven.compiler.source>1.8</maven.compiler.source>
        <maven.compiler.target>1.8</maven.compiler.target>
        <asciidoctor.maven.plugin.version>2.2.2</asciidoctor.maven.plugin.version>
        <asciidoctorj.epub.version>1.5.1</asciidoctorj.epub.version>
        <asciidoctorj.version>2.5.5</asciidoctorj.version>
        <jruby.version>9.3.4.0</jruby.version>
```

`mvn -e`


```
[ERROR] Failed to execute goal org.asciidoctor:asciidoctor-maven-plugin:3.0.0-SNAPSHOT:process-asciidoc (generate-spine-kf8) on project asciidoctor-epub-example: Execution generate-spine-kf8 of goal org.asciidoctor:asciidoctor-maven-plugin:3.0.0-SNAPSHOT:process-asciidoc failed: org.jruby.exceptions.SystemCallError: (EBADF) Bad file descriptor - /bin/sh -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.asciidoctor:asciidoctor-maven-plugin:3.0.0-SNAPSHOT:process-asciidoc (generate-spine-kf8) on project asciidoctor-epub-example: Execution generate-spine-kf8 of goal org.asciidoctor:asciidoctor-maven-plugin:3.0.0-SNAPSHOT:process-asciidoc failed: org.jruby.exceptions.SystemCallError: (EBADF) Bad file descriptor - /bin/sh
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:375)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution generate-spine-kf8 of goal org.asciidoctor:asciidoctor-maven-plugin:3.0.0-SNAPSHOT:process-asciidoc failed: org.jruby.exceptions.SystemCallError: (EBADF) Bad file descriptor - /bin/sh
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:148)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.asciidoctor.jruby.internal.AsciidoctorCoreException: org.jruby.exceptions.SystemCallError: (EBADF) Bad file descriptor - /bin/sh
    at org.asciidoctor.jruby.internal.JRubyAsciidoctor.convertFile (JRubyAsciidoctor.java:402)
    at org.asciidoctor.jruby.internal.JRubyAsciidoctor.convertFile (JRubyAsciidoctor.java:368)
    at org.asciidoctor.maven.AsciidoctorMojo.convertFile (AsciidoctorMojo.java:451)
    at org.asciidoctor.maven.AsciidoctorMojo.processSources (AsciidoctorMojo.java:250)
    at org.asciidoctor.maven.AsciidoctorMojo.processAllSources (AsciidoctorMojo.java:166)
    at org.asciidoctor.maven.AsciidoctorMojo.execute (AsciidoctorMojo.java:156)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.jruby.exceptions.SystemCallError: (EBADF) Bad file descriptor - /bin/sh
    at org.jruby.RubyProcess.spawn (org/jruby/RubyProcess.java:1671)
    at org.jruby.RubyKernel.spawn (org/jruby/RubyKernel.java:1723)
    at RUBY.popen_run (uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/open3.rb:213)
    at RUBY.popen3 (uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/open3.rb:101)
    at RUBY.capture3 (uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/open3.rb:281)
    at RUBY.distill_epub_to_mobi (uri:classloader:/gems/asciidoctor-epub3-1.5.1/lib/asciidoctor-epub3/converter.rb:1679)
    at RUBY.write (uri:classloader:/gems/asciidoctor-epub3-1.5.1/lib/asciidoctor-epub3/converter.rb:42)
    at RUBY.write (uri:classloader:/gems/asciidoctor-2.0.17/lib/asciidoctor/document.rb:988)
    at RUBY.convert (uri:classloader:/gems/asciidoctor-2.0.17/lib/asciidoctor/convert.rb:124)
    at RUBY.convert_file (uri:classloader:/gems/asciidoctor-2.0.17/lib/asciidoctor/convert.rb:190)
    at org.jruby.RubyIO.open (org/jruby/RubyIO.java:1148)
    at RUBY.convert_file (uri:classloader:/gems/asciidoctor-2.0.17/lib/asciidoctor/convert.rb:190)
[ERROR] 
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException

```

--- a/asciidoctor-epub-example/pom.xml
+++ b/asciidoctor-epub-example/pom.xml
@@ -50,10 +50,12 @@
                     </dependency>
                 </dependencies>
                 <configuration>
-                    <sourceDirectory>src/docs/asciidoc</sourceDirectory>
-                    <attributes>
-                        <sourcedir>${project.build.sourceDirectory}</sourcedir>
-                    </attributes>
+                    <backend>epub3</backend>
+                    <sourceDocumentName>spine.adoc</sourceDocumentName>
+<!--                    <sourceDirectory>src/docs/asciidoc</sourceDirectory>-->
+<!--                    <attributes>-->
+<!--                        <sourcedir>${project.build.sourceDirectory}</sourcedir>-->
+<!--                    </attributes>-->
                     <resources>
                         <resource>
                             <directory>.</directory>
@@ -71,67 +73,21 @@
                         <goals>
                             <goal>process-asciidoc</goal>
                         </goals>
-                        <configuration>
-                            <backend>epub3</backend>
-                            <sourceDocumentName>spine.adoc</sourceDocumentName>
-                        </configuration>
                     </execution>
-
-
-                    <!-- for converting asciidoc to a mobi format, asciidoc performs 2 steps. -->
-                    <!-- first, it generates a xxxx-kf8.epub file, which is then, in the second step, processed in order
-                         to create xxxx.mobi file. -->
-
-                    <!-- general information: epub and mobi generation differs only in the attribute 'ebook-format',
-                         which is set to 'epub3' for epub and to kf8 in order to generate a special annotated epub3
-                         document which is suitable for kindlegen -->
-
-                    <!-- per default, this attribute is set to 'epub3'. -->
-                    <!-- you can set this attribute either in the corresponding .adoc file, or provide it like showed
-                         in this execution block below -->
+                    <!-- For testing and troubleshooting you can see the
+                     contained resources in an extracted structure -->
                     <execution>
-                        <id>generate-spine-kf8</id>
+                        <id>generate-extracted-epub</id>
                         <phase>generate-resources</phase>
                         <goals>
                             <goal>process-asciidoc</goal>
                         </goals>
                         <configuration>
-                            <backend>epub3</backend>
-                            <sourceDocumentName>spine.adoc</sourceDocumentName>
                             <attributes>
-                                <source-highlighter>rouge</source-highlighter>
-                                <ebook-format>kf8</ebook-format>
+                                <ebook-extract/>
                             </attributes>
                         </configuration>
                     </execution>
-
-
-                    <!-- attention: if there are problems calling kindlegen (see discussion here https://github.com/asciidoctor/asciidoctor-maven-examples/pull/68) -->
-                    <!-- then you can call kindlegen directly via maven like showed below -->
-                    <!---
-
-                </executions>
-            </plugin>
-            <plugin>
-                <groupId>org.codehaus.mojo</groupId>
-                <artifactId>exec-maven-plugin</artifactId>
-                <version>1.6.0</version>
-                <executions>
-                    <execution>
-                        <id>generate-mobi-file</id>
-                        <phase>generate-resources</phase>
-                        <goals>
-                            <goal>exec</goal>
-                        </goals>
-                        <configuration>
-                            <workingDirectory>${project.build.directory}/generated-docs/</workingDirectory>
-                            <executable>${path.to.kindlegen}</executable>
-                            <arguments>
-                                <argument>spine.epub</argument>
-                            </arguments>
-                        </configuration>
-                    </execution>
-                    -->
                 </executions>
             </plugin>
         </plugins>

--- a/tests/src/test/java/org/asciidoctor/maven/examples/AsciidoctorEpubTest.java
+++ b/tests/src/test/java/org/asciidoctor/maven/examples/AsciidoctorEpubTest.java
@@ -2,28 +2,56 @@ package org.asciidoctor.maven.examples;
 
 import org.asciidoctor.maven.examples.tests.MavenProject;
 import org.asciidoctor.maven.examples.tests.MavenTest;
-import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 
 import java.io.File;
+import java.util.function.Predicate;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
-@Disabled("issue-141")
 @MavenTest(projectPath = "../asciidoctor-epub-example")
 class AsciidoctorEpubTest {
 
     private MavenProject mavenProject;
 
     @Test
-    void shouldGenerateEpubFiles() {
+    void shouldGenerateEpubFile() {
         File epub = mavenProject.getTarget(generatedDocs("spine.epub"));
-        assertThat(epub)
-            .isNotEmpty();
 
-        File epubKf8 = mavenProject.getTarget(generatedDocs("spine-kf8.epub"));
-        assertThat(epubKf8)
-            .isNotEmpty();
+        assertThat(epub)
+            .isNotEmpty()
+            .isFile();
+    }
+
+    @Test
+    void shouldGenerateExtractedDir() {
+        File extractedDir = mavenProject.getTarget(generatedDocs("spine"));
+
+        assertThat(extractedDir)
+            .isDirectory()
+            .isDirectoryContaining(directory("EPUB"))
+            .isDirectoryContaining(directory("META-INF"))
+            .isDirectoryContaining(file("mimetype"));
+
+        assertThat(new File(extractedDir, "EPUB"))
+            .isDirectory()
+            .isDirectoryContaining(file("_chapter_1.xhtml"))
+            .isDirectoryContaining(file("_chapter_2.xhtml"))
+            .isDirectoryContaining(file("_chapter_3.xhtml"));
+
+        assertThat(new File(extractedDir, "EPUB/images"))
+            .isDirectory()
+            .isDirectoryContaining(file("sunset.jpg"));
+
+        extractedDir.toPath().resolve("");
+    }
+
+    private static Predicate<File> directory(String name) {
+        return file -> file.isDirectory() && file.getName().equals(name);
+    }
+
+    private static Predicate<File> file(String name) {
+        return file -> file.isFile() && file.getName().equals(name) && file.length() > 0;
     }
 
     private String generatedDocs(String filename) {

